cmd/gdeploy/commands/provider/add.go

```diff
@@ -27,69 +30,92 @@ var addCommand = cli.Command{
 		}
 
 		r := providerregistry.Registry()
-		p := survey.Select{
-			Message: "What are you trying to grant access to?",
-			Options: r.CLIOptions(),
+
+		uses := c.String("uses")
+		var provider *providerregistry.RegisteredProvider
+
+		if uses == "" {
+			p := survey.Select{
+				Message: "What are you trying to grant access to?",
+				Options: r.CLIOptions(),
+			}
+			var chosen string
+			err = survey.AskOne(&p, &chosen)
+			if err != nil {
+				return err
+			}
+			uses, provider, err = r.FromCLIOption(chosen)
+			if err != nil {
+				return err
+			}
+			clio.Info("Follow the documentation for setting up the %s provider here: https://docs.commonfate.io/granted-approvals/providers/%s", provider.Description, provider.DefaultID)
+		} else {
+			provider, err = r.LookupByUses(uses)
+			if err != nil {
+				return err
+			}
 		}
-		var chosen string
-		err = survey.AskOne(&p, &chosen)
-		if err != nil {
-			return err
+
+		id := c.String("id")
```

Comment:
Given the ID does show up in the UI for users, should we be keeping the significance here?
Or perhaps adding a friendly Name field?

Reply:
name field could be good - we could make it optional too. My thinking on the changes to “id” in this CLI command was to try and reduce the cognitive overhead for users by eliminating the question prompting for the provider ID. It should only be needed in advanced use cases where there are multiple copies of one provider installed.

I think showing the ID is useful (I know this isn’t related to your question but just sharing my POV here) as it makes it easy to see which access rules refer to which provider.